src/tools/tidy/src/ui_tests.rs

```diff
@@ -1,9 +1,42 @@
-//! Tidy check to ensure that there are no stray `.stderr` files in UI test directories.
+//! Tidy check to ensure:
+//! - the number of entries in each directory must be less than 1000
+//! - there are no stray `.stderr` files
+//! in UI test directories.
 
 use std::fs;
 use std::path::Path;
 
+const ENTRY_LIMIT: usize = 1000;
+
+fn check_entries(path: &Path, bad: &mut bool) {
+    let dirs = walkdir::WalkDir::new(&path.join("test/ui"))
+        .into_iter()
+        .filter_entry(|e| e.file_type().is_dir());
+    // FIXME: We should not allow them ideally.
+    let allowed_paths = ["test/ui", "test/ui/issues"];
```

Comment:
Could you set different entry limits for these directories instead? (Equal to the current number of entries in them.)

Reply:
Sounds like a better solution! Addressed reviews via 831f5ab.

Reply:
I mean 1000 is still the reasonable limit for all directories except for "test/ui" and "test/ui/issues", so we can use three different limits - 1567 for "test/ui", 2816 for "test/ui/issues" and 1000 for everything else.

Reply:
Oh, I got it wrong. Set limits that are just a little more than the current number of entries to avoid failures on merging.